### ISSUE TITLE
"Open folder" from code-server hangs

### ISSUE THREAD

**Issue by code-asher** [coder/coder#3316]
The way "open folder" is implemented is to set `window.location.href` with the `folder` query variable.

Manually editing the URL, manually calling `window.location.href`, or even just reloading the page can also trigger the hang.  It may take several attempts or waiting a bit (at least 7-8 seconds seems to be the ticket) before doing the navigation.  The resulting request will be pending forever.

This is on Coder v0.8.1-devel+6e63487b running in an image with the `coder` template.  It happens on both Firefox and Chromium.

I was not able to reproduce it outside of Coder.

**Comment by code-asher** [comment]
Interestingly it does not happen if I forward the port through SSH and access code-server that way.  I guess this implicates the networking or the proxy in some way?

**Comment by bpmct** [comment]
I believe #2779 is coming in soon as an environment variabke. Will be interesting to test if this still happens once that is done.

**Comment by kylecarbs** [comment]
Closing in favor of #2779.
